### DISCUSSION TITLE
File versioning

### DISCUSSION THREAD

**Discussion by PHPGuus** [spatie/laravel-medialibrary#2611] [Ideas] [answered]
Hi there,

is there appetite for creating a feature in the laravel-medialibrary that will enable file versioning?

I would see this working as any of the starter methods (`addMedia` and friends) checking whether a file with the given name already exists in the database for the given model, and if so, makes sure that it keeps the previous version in a special "versions" (configurable, of course) subdirectory (much like the subdirectories for media conversions). The main file location would then always have the latest version, but people can retrieve previous versions if they wanted to.

Enabling this feature would be configurable (off, by default) through the `media-library.php` config file.

Retrieving a specific version of a media would require to add a new argument (say, `$version`) to the `getUrl` and other retriever functions to allow a user to (try to) retrieve a particular version of the file.

I would even think, a class `FileVersionGenerator` could be provided by default that would create version numbers for the versioning implementation. People can then exchange that class with their own implementation, if the standard incremental version numbers (1, 2, etc) are not wanted.

I am more than happy to write this functionality and do a PR etc. I just wanted to see whether the maintainers would be willing to accept this feature into the media-library.

Best regards,
Guus

**Comment by freekmurze** [comment, marked as answer]
Not too many people has asked for this functionality. I'm a bit hesitant of adding a feature like this as I want to keep this package as simple as possible.

Please do send a PR for this for us to review. If it's something we're willing to maintain we accept it. And if not, you can probably still use the code for your own project.

**Comment by PHPGuus** [comment]
Hi Freek,

OK will do.

Thank you!

**Comment by Forsakenrox** [comment]
I request this feature too

**Reply by PHPGuus** [reply]
Means I've got to find my writer's pen, eh :-)

**Comment by dgillier** [comment]
Any update on this feature ?
Thanks,
Denis

**Comment by sekrap23** [comment]
Would love to see this feature done!
Any update on its progress?
Thanks

**Reply by parallels999** [reply]
>Any update on its progress?

No, feel free to make it possible

**Reply by sekrap23** [reply]
@parallels999 I didn't ask you

**Reply by parallels999** [reply]
lol, well, I inform you that this is public, 
and this is my backup account

**Reply by sekrap23** [reply]
1/ I'm aware.
2/ Congratulations.

My initial comment was friendly and didn't warrant a passive aggressive response. I'm actually more than happy to contribute to this feature because it would be a great addition, but if there's someone already working on it then surely it's ok if I ask about it?

**Reply by sekrap23** [reply]
Also, a (seemingly) random account replying to my comment with no previous interaction with the thread looks like someone just looking for trouble, hence my response. Maybe you should have commented with your main account and this would have went a lot better 🙂

**Reply by parallels999** [reply]
Not working on it, look at the fork, i didn't even check it to be honest, feel free to contribute and i will review your PR

**Reply by jose123v** [reply]
>Would love to see this feature done!
Any update on its progress?
Thanks

>@parallels999 I didn't ask you

If you don't tag anyone, the question is for everyone who reads it.

>Also, a (seemingly) random account

"random account" has more open source contributions than you

**Reply by sekrap23** [reply]
Thanks @jose123v, really helpful, have a great day!

**Reply by fredsal** [reply]
> I'm actually more than happy to contribute to this feature because it would be a great addition

@SekrapElad hi, any update on its progress?

**Reply by sekrap23** [reply]
@fredsal No, feel free to make it possible

**Reply by fredsal** [reply]
>I'm actually more than happy to contribute to this feature because it would be a great addition

@SekrapElad If I knew how to do it I would, but since you offer to do it I thought you are making it, and as you say "if there's someone already working on it then surely it's ok if I ask about it"

>@fredsal No, feel free to make it possible

any guide or indication, where I can see where you stayed to see if I can add any progress

**Reply by sekrap23** [reply]
@fredsal Yawnnnnn 🥱 I've much better things to do rather than reply back to your nonsense. Enjoy the rest of your day commenting on threads that have nothing to do with you, sounds really exciting 😘

**Reply by fredsal** [reply]
lol, demanding something is easy, doing it is not, enjoy the rest of your life demanding on threads that have nothing to do with you, sounds really exciting 😘, bye
